src/pecora.jl

```diff
@@ -219,7 +219,7 @@ function pecora(
     end
     undersampling && metric ≠ Chebyshev() && error("Chebyshev metric required for undersampling")
     vspace = genembed(s, τs, js)
-    vtree = KDTree(vspace.data, metric)
+    vtree = KDTree(vspace.data[1:end-maximum(delays)], metric)

```

Comment:
Here I constrained the total number of points (including the nearest neighbours) with respect to the maximum considered delay time.